Mathlib/Algebra/Algebra/Subalgebra/Basic.lean

```diff
@@ -768,6 +768,28 @@ theorem iSup_toSubsemiring {ι : Sort*} [Nonempty ι] (S : ι → Subalgebra R A
     (iSup S).toSubsemiring = ⨆ i, (S i).toSubsemiring := by
   simp only [iSup, Set.range_nonempty, sSup_toSubsemiring, ← Set.range_comp, Function.comp_def]
 
+lemma mem_iSup_of_mem {ι : Sort*} {S : ι → Subalgebra R A} (i : ι) {x : A} (hx : x ∈ S i) :
+    x ∈ iSup S :=
+  le_iSup S i hx
+
+@[elab_as_elim]
+lemma iSup_induction {ι : Sort*} (S : ι → Subalgebra R A) {motive : A → Prop}
+    {x : A} (mem : x ∈ ⨆ i, S i)
+    (basic : ∀ i, ∀ a ∈ S i, motive a)
+    (zero : motive 0) (one : motive 1)
+    (add : ∀ a b, motive a → motive b → motive (a + b))
+    (mul : ∀ a b, motive a → motive b → motive (a * b))
+    (algebraMap : ∀ r, motive (algebraMap R A r)) : motive x := by
```

Comment:
Can you add the version analogous to [`Submodule.iSup_induction'`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Span/Basic.html#Submodule.iSup_induction') too?

Reply:
done